dali/pipeline/data/tensor.h

```diff
@@ -176,6 +181,10 @@ class Tensor : public Buffer<Backend> {
    *
    * If the input does not store any data, shares_data_ is left
    * as false.
+   *
+   * The total size after using `set_type` and `Resize` must match
```

Comment:
Do we need that info here? It seems to be not relevant when `ShareData` is called.

Reply:
The comment explains the semantics of what will be happening if you actually share data. Or do you think it is better to leave it without any info and just throw errors when users misbehave?

Reply:
Maybe rephrasing it to:
```
* After calling this function any following call to `set_type` and `Resize`
* must match the total size of underlying allocation (`num_bytes_`) of
* shared data or call will fail.
* It can be set to 0 as intermediate step.
```

Reply:
Done everywhere.